### DISCUSSION TITLE
How to properly obtain and use MHFP Fingerprint?

### DISCUSSION THREAD

**Discussion by jshan2020-33** [rdkit/rdkit#5556] [Q&A]
The code used to get MHFP is as follows:

from rdkit import Chem
from rdkit.Chem.rdMHFPFingerprint import MHFPEncoder

mol = Chem.MolFromSmiles('OCC(=O)CCCN')

a = MHFPEncoder()
MHFP = MHFPEncoder.EncodeMol(a, mol)
MHFP_vector = [int(i) for i in MHFP]

If the acquisition method is correct, can MHFP be directly input into the machine learning model as a feature without conversion?
The obtained value of MHFP is not 0/1, but a very large value, as shown in the following figure:
![image](https://user-images.githubusercontent.com/81633044/189532759-19be3802-2b2b-4e1f-8555-af9b6670754f.png)

-----------------
Python: 3.7.0
RDKit: 2020.09.1

**Comment by j-adamczyk** [comment]
Those are MinHash raw hash values. They are meant to be used with Jaccard similarity, using built-in method for fast comparisons. See: https://iwatobipen.wordpress.com/2018/10/13/new-fingerprint-minhash-fingerprint-rdkit-chemoinformatics/.

I'm also looking for a way to use this in generic ML applications. Maybe just use modulo operation?

**Comment by Olabisi-Aishat-Bello** [comment]
Based on the original [paper](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-018-0321-8), the SECFP4/6 seems to do just that, and this is available with RDKIT

"a fingerprint variant on MHFP’s SMILES-based circular substructure hashing scheme, folded by the same modulo n operation that is used by ECFP, was compared to both minhashed MHFP and folded ECFP with r ∈ {2, 3} and D = 2048 (Fig.  7). We denoted this variant SECFP
(SMILES extended connectivity fingerprint)."
